app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -647,6 +647,8 @@ private void updateSearchUI() {
         outdoorDirections.deleteDirection();
         if (location != null && destination != null) {
             outdoorDirections.requestDirections();
+            GlobalPathFinder globalPathFinder = new GlobalPathFinder((MainActivity)getActivity(),location,destination);
+            globalPathFinder.run();
```

Comment:
Additionally, this should be in the MainActivity so that it can be accessed by all fragments.